### ISSUE TITLE
SyntaxError: Invalid RegExp: Invalid escape, js engine: hermes

### ISSUE THREAD

**Issue by Albert-Gao** [StefanTerdell/zod-to-json-schema#129]
## Bug Description

a regexp from this lib triggers this error on Hermes(React native).

```javascript
// node_modules/zod-to-json-schema/dist/cjs/parsers/string.js
exports.zodPatterns = {
  // remove other codes
  emoji: RegExp("^(\\p{Extended_Pictographic}|\\p{Emoji_Component})+$", "u"),
};
```

![image](https://github.com/user-attachments/assets/71af5341-80aa-4e74-8816-47ad3ba36ae8)

- "expo": "~51.0.21",
- "react-native": "0.74.3"

## Steps To Reproduce

a repo to reproduce it.

https://github.com/Albert-Gao/rn-invalid-regexp

## The Expected Behavior

It should work. (Works on JSC)

## More

It seems to be a Herms issue according to this: https://github.com/facebook/hermes/issues/1027
So I first created an issue on the Hermes side: https://github.com/facebook/hermes/issues/1460
But if it is really hard to solve on the engine side, I wonder is there a way we can tweak this regexp to make it compatible with Hermes?

**Comment by Albert-Gao** [comment]
seems it will be solved in the future version of Hermes, so close this one :)

**Comment by StefanTerdell** [comment]
@Albert-Gao This is now avoided with lazy-loading. Released in 3.23.2.

**Comment by Albert-Gao** [comment]
@StefanTerdell thanks, i tested, works like a charm with Hermes now :)
